pallets/ah-migrator/src/lib.rs

```diff
@@ -182,6 +189,56 @@ pub mod pallet {
 	pub type RcAccounts<T: Config> =
 		StorageMap<_, Twox64Concat, T::AccountId, RcAccountFor<T>, OptionQuery>;
 
+	/// Amount of balance that was reserved for winning a lease auction.
+	///
+	/// `unreserve_lease_deposit` can be permissionlessly called once the block number passed to
+	/// unreserve the deposit. It is implicitly called by `withdraw_crowdloan_contribution`.
+	///  
+	/// The account here can either be a crowdloan account or a solo bidder. If it is a crowdloan
+	/// account, then the summed up contributions for it in the contributions map will equate the
+	/// reserved balance here.
+	///
+	/// The keys are as follows:
+	/// - Block number after which the deposit can be unreserved.
+	/// - The account that will have the balance unreserved.
+	/// - The para_id of the lease slot.
+	/// - The balance to be unreserved.
+	#[pallet::storage]
+	pub type RcLeaseReserve<T: Config> = StorageNMap<
+		_,
+		(
+			NMapKey<Twox64Concat, BlockNumberFor<T>>,
+			NMapKey<Twox64Concat, T::AccountId>,
+			NMapKey<Twox64Concat, ParaId>,
+		),
+		BalanceOf<T>,
+		OptionQuery,
+	>;
+
+	/// Amount of balance that a contributor made towards a crowdloan.
+	///
+	/// `withdraw_crowdloan_contribution` can be permissionlessly called once the block number
+	/// passed to unlock the balance for a specific account.
+	///
+	/// The keys are as follows:
+	/// - Block number after which the balance can be unlocked.
+	/// - The account that made the contribution.
+	/// - The para_id of the crowdloan.
+	///
+	/// The value is (fund_pot, balance). The contribution pot is the second key in the
+	/// `RcCrowdloanContribution` storage.
+	#[pallet::storage]
+	pub type RcCrowdloanContribution<T: Config> = StorageNMap<
+		_,
```

Comment:
@muharem the only thing to look at in the current state would be the data structures.